Fix footer heading semantic consistency and alignment

sf-cboscenco opened the pull request:
- Change "Be the first to know" heading from h1 to h2 for proper semantic hierarchy
- Add responsive paddingTop to subscribeHeading (desktop: 3, mobile: 0) to align with other footer grid headings on desktop while maintaining clean positioning on mobile

The h1 usage was semantically incorrect as footer sections should be h2 level, and the padding ensures visual alignment with LinksList headings in the 4-column desktop grid layout.

<!--- Provide a short summary of your changes in the Title field above -->

# Description

<!--- A longer summary of your changes, including: a description of the issue that you’re addressing, a list of required dependencies (if applicable), and any other relevant context. -->

# Types of Changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->

- [x] **Bug fix** (non-breaking change that fixes an issue)
- [ ] **New feature** (non-breaking change that adds functionality)
- [ ] **Documentation update**
- [ ] **Breaking change** (could cause existing functionality to not work as expected)
- [ ] **Other changes** (non-breaking changes that does not fit any of the above)

> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

Files changed:
- packages/template-retail-react-app/app/components/footer/index.jsx
- packages/template-retail-react-app/app/theme/components/project/footer.js

# How to Test-Drive This PR

- (step1)

# Checklists

<!--- Enter an `x` in all the boxes that apply. -->
<!--- If you’re unsure about any of these, don’t hesitate to ask. We’re here to help! -->

## General

- [x] Changes are covered by test cases
- [x] CHANGELOG.md updated with a short description of changes (_not_ required for documentation updates)

## Accessibility Compliance

You must check off all items in **one** of the follow two lists:

- [ ] There are no changes to UI

_or..._

- [ ] Changes were tested with a Screen Reader (iOS VoiceOver or Android Talkback) and had no issues
- [ ] Changes comply with [WCAG 2.0 guidelines levels A and AA](https://www.wuhcag.com/wcag-checklist/)
- [ ] Changes to common UI patterns and interactions comply with [WAI-ARIA best practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

## Localization

- [x] Changes include a UI text update in the Retail React App (which requires translation)

sf-cboscenco committed:
Fix footer heading semantic consistency and alignment

- Change "Be the first to know" heading from h1 to h2 for proper semantic hierarchy
- Add responsive paddingTop to subscribeHeading (desktop: 3, mobile: 0) to align with other footer grid headings on desktop while maintaining clean positioning on mobile

The h1 usage was semantically incorrect as footer sections should be h2 level, and the padding ensures visual alignment with LinksList headings in the 4-column desktop grid layout.

cc-prodsec commented:
### :white_check_mark: **Snyk checks have passed. No issues have been found so far.**

| Status | Scanner | ![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/severity-critical.svg) Critical | ![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/severity-high.svg) High | ![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/severity-medium.svg) Medium | ![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/severity-low.svg) Low | Total (0) |
|---|---|---|---|---|---|---|
| :white_check_mark: | Licenses| 0 | 0 | 0 | 0 | [ 0 issues](https://app.snyk.io/org/cc-mobify-sdk/pr-checks/b3105261-e402-4109-9e35-8d03c5276acc/license?refs=c287be46d8b31e28c73e650bd65751790ff54dc0&source=prChecksComment) |
| :white_check_mark: | Open Source Security| 0 | 0 | 0 | 0 | [ 0 issues](https://app.snyk.io/org/cc-mobify-sdk/pr-checks/b3105261-e402-4109-9e35-8d03c5276acc?refs=c287be46d8b31e28c73e650bd65751790ff54dc0&source=prChecksComment) |

:computer: Catch issues earlier using the plugins for [VS Code](https://marketplace.visualstudio.com/items?itemName=snyk-security.snyk-vulnerability-scanner), [JetBrains IDEs](https://plugins.jetbrains.com/plugin/10972-snyk-security), [Visual Studio](https://marketplace.visualstudio.com/items?itemName=snyk-security.snyk-vulnerability-scanner-vs-2022), and [Eclipse](https://marketplace.eclipse.org/content/snyk-security#external-install-button).

ddiazccrz commented:
I think this might need a changelog since its going in to develop as a bug fix. @alexvuong @bfeister

alexvuong commented:
Please add a changelog for this

sf-cboscenco commented:
> Please add a changelog for this

What version should I use?

alexvuong commented:
Ci is failing because these changes has caused a11y violation. 
```
{
      +    "id": "page-has-heading-one",
      +    "impact": "moderate",
      +    "description": "Ensure that the page, or at least one of its frames contains a level-one heading",
      +    "help": "Page should contain a level-one heading",
      +    "helpUrl": "https://dequeuniversity.com/rules/axe/4.10/page-has-heading-one?application=playwright",
      +    "nodes": [
      +      {
      +        "html": "<html lang=\"en-US\" class=\"...\"   >",
      +        "failureSummary": "Fix all of the following:\n  Page must have a level-one heading",
      +        "target": [
      +          "html"
      +        ]
      +      }
```

alexvuong commented:
> > Please add a changelog for this
> 
> What version should I use?

The up-coming release. It is current 8.2.0-dev

sf-cboscenco committed:
Fix accessibility compliance by adding proper h1 headings to checkout pages to resolve the page-has-heading-one accessibility rule violation.

sf-cboscenco committed:
Added changelog entry.

sf-cboscenco committed:
Fix more a11y issues on various pages

sf-cboscenco committed:
Update compiled translations with new accessibility h1 heading keys

- Regenerated compiled translation files to include new h1 heading keys
- Ensures all accessibility translation keys are available at runtime
- Resolves Lerna EUNCOMMIT error for publishing

sf-cboscenco committed:
Complete translation build for accessibility h1 keys

- Updated source translation files (en-US, en-GB) with extracted default messages
- Regenerated pseudo-locale (en-XA) with new accessibility keys
- Ensures all translation files are in sync for Lerna publishing
- Resolves remaining EUNCOMMIT errors

sf-cboscenco commented:
> Ci is failing because these changes has caused a11y violation.
> 
> ```
> {
>       +    "id": "page-has-heading-one",
>       +    "impact": "moderate",
>       +    "description": "Ensure that the page, or at least one of its frames contains a level-one heading",
>       +    "help": "Page should contain a level-one heading",
>       +    "helpUrl": "https://dequeuniversity.com/rules/axe/4.10/page-has-heading-one?application=playwright",
>       +    "nodes": [
>       +      {
>       +        "html": "<html lang=\"en-US\" class=\"...\"   >",
>       +        "failureSummary": "Fix all of the following:\n  Page must have a level-one heading",
>       +        "target": [
>       +          "html"
>       +        ]
>       +      }
> ```

Yeah, this was because of some old issues related to pages not having a h1 tag, I fixed all these pages.

sf-cboscenco committed:
Merge branch 'develop' into t/cc-sharks/W-19721000

sf-cboscenco committed:
fix: resolve e2e test failures for DNT, PLP and checkout a11y tests

- Fix DNT test selector to target button instead of heading role
  The test was failing due to strict mode violation where multiple
  'My Account' headings existed (h1 on page + h2 in accordion).
  Changed selector from getByRole('heading') to getByRole('button')
  to specifically target the accordion button.

- Add h1 heading to Product Listing Page for accessibility
  Changed page header from h2 to h1 to ensure PLP has proper
  semantic heading structure and resolves page-has-heading-one
  violation for both mobile and desktop a11y tests.

- Update checkout a11y snapshots to reflect h1 heading fix
  The checkout page now has proper h1 heading (fixed elsewhere),
  so removed page-has-heading-one violation from test snapshots
  for both guest and registered user flows.

Fixes test failures:
- [mobile-chrome] dnt.spec.js - Shopper can use consent tracking form
- [a11y-mobile-slas-private-client] PLP accessibility tests
- [a11y-desktop-slas-private-client] PLP accessibility tests
- [a11y-desktop-slas-private-client] Checkout accessibility tests (guest + registered)

sf-cboscenco committed:
fix: update DNT test selector and checkout step-1 a11y snapshots

- Make DNT test selector more specific by targeting 'My Account chevron-down'
  to uniquely identify the accordion button and avoid matching the header icon button

- Update checkout step-1 accessibility snapshots for guest and registered users
  to remove page-has-heading-one violation (checkout now has proper h1 heading)

sf-cboscenco committed:
fix: update remaining checkout a11y snapshots (step-2, step-3, step-4)

- Update guest checkout step-3 snapshot to remove page-has-heading-one violation
- Update registered checkout step-2 and step-3 snapshots
- Update guest checkout step-4-order-confirmation snapshot

All checkout steps now properly reflect that pages have h1 headings.

sf-cboscenco committed:
fix: update login test selectors to use getByRole for button

- Fixed 3 failing login tests where 'Sign In' text matched both h1 heading and button
- Changed from getByText(/sign in/i) to getByRole('button', {name: /sign in/i})
- Follows same pattern as DNT test fix for specificity

sf-cboscenco committed:
fix: update test expectations for account page headings

- Updated use-auth-modal test to expect 3 'My Account' headings (was 2)
- Updated registration test to expect 3 'My Account' headings (was 2)
- Fixed registration test button selector to use getByRole for specificity
- Account page now has h1 (sr-only), h2 (accordion), and h2 (sidebar)

patricksullivansf reviewed (approved):
Thank you for the work involved  in the clean up here!
